core/src/main/scala/kafka/server/ForwardingManager.scala

```diff
@@ -77,7 +80,8 @@ class ForwardingManagerImpl(
 
   override def forwardRequest(
     request: RequestChannel.Request,
-    responseCallback: AbstractResponse => Unit
+    responseCallback: AbstractResponse => Unit,
```

Comment:
I think we could generalize the response callback and avoid the additional callback argument. Perhaps we could return `Either[Response, Exception]`. Then in `KafkaApis`, we can log an error in the exceptional case and close the connection.